source/kernel/src/registry/mod.rs

```diff
@@ -451,7 +460,29 @@ impl Registry {
 
 // Envelope
 
+impl<P> OpenEnvelope<P> {
+    pub fn fill(self, contents: P) -> Envelope<P> {
+        Envelope {
+            body: contents,
+            service_id: self.service_id,
+            client_id: self.client_id,
+            request_id: self.request_id,
+        }
+    }
+}
+
 impl<P> Envelope<P> {
+    // TODO: constrain types
```

Comment:
I can remove this, it's now private and `Message::split` does constrain types appropriately.